src/components/routes/RequireGuest.tsx

```
@@ -0,0 +1,16 @@
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { useAuth } from '@/hooks/useAuth';
+import { PATH } from '@/utils/path';
+
+export default function RequireGuest() {
+  const { isAuthenticated } = useAuth();
+  const location = useLocation();
+  const from = (location.state as any)?.from?.pathname as string | undefined;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)